checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakVisitor.java

```diff
@@ -392,13 +363,20 @@ private void checkOwningField(Element field) {
     }
 
     // This value is side-effected.
-    List<String> unsatisfiedMustCallObligationsOfOwningField =
-        rlTypeFactory.getMustCallValues(field);
+    List<String> mustCallObligationsOfOwningField = rlTypeFactory.getMustCallValues(field);
 
-    if (unsatisfiedMustCallObligationsOfOwningField.isEmpty()) {
+    if (mustCallObligationsOfOwningField.isEmpty()) {
       return;
     }
 
+    Set<DestructorObligation> unsatisfiedMustCallObligationsOfOwningField =
+        mustCallObligationsOfOwningField.stream()
+            .flatMap(
+                method ->
+                    Arrays.stream(MustCallConsistencyAnalyzer.MethodExitKind.values())
+                        .map(exitKind -> new DestructorObligation(method, exitKind)))
+            .collect(Collectors.toCollection(LinkedHashSet::new));
```

Comment:
@mernst is there a rule (or rule of thumb) around use of stream APIs in the Checker Framework?

Reply:
I've gone ahead and removed the use of streams. This is a case where the code is actually clearer with normal for-loops.